### ISSUE TITLE
ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected

### ISSUE THREAD

**Issue by sprabhuonline** [edenhill/librdkafka#1973]
Read the FAQ first: https://github.com/edenhill/librdkafka/wiki/FAQ

Description
===========
If there is no data available on producer we are getting following error.

16-08-2018 17:28:51.038 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
16-08-2018 18:53:51.275 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
16-08-2018 19:32:48.091 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
16-08-2018 20:23:51.325 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
16-08-2018 21:58:51.132 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
16-08-2018 22:52:48.139 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
16-08-2018 23:02:48.410 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 00:02:48.113 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 00:28:51.045 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 01:38:51.076 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 02:48:51.187 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 04:02:48.214 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 04:12:48.317 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 05:02:48.286 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 05:32:48.116 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected
17-08-2018 06:13:51.046 ERROR - ERROR (Local: Broker transport failure): kafka1:9092/bootstrap: Receive failed: Disconnected

How to reproduce
================
Send some data on producer and stop for some long duration like 2 hrs. you can see this error

**IMPORTANT**: Always try to reproduce the issue on the latest released version (see https://github.com/edenhill/librdkafka/releases), if it can't be reproduced on the latest version the issue has been fixed.

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [x] librdkafka version (release number or git tag): `<REPLACE with e.g., v0.10.5 or a git sha. NOT "latest" or "current">`
 - [ ] Apache Kafka version: `<REPLACE with e.g., 0.10.2.3>`
 - [ ] librdkafka client configuration: `<REPLACE with e.g., message.timeout.ms=123, auto.reset.offset=earliest, ..>`
 - [ ] Operating system: `<REPLACE with e.g., Centos 5 (x64)>`
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by rnpridgeon** [comment]
Hello @sprabhuonline , 

Please review the following section of the librdkafka FAQ

https://github.com/edenhill/librdkafka/wiki/FAQ#why-am-i-seeing-receive-failed-disconnected

TLDR: Broker's are configured with an idle connection reaper which is causing the disconnect; by default the idle connection sweep executes every 10 minutes. Once librdkafka realizes the connection is no longer valid it reestablishes a connection and the cycle starts again. You may find that this happens even when you are producing messages as well. This is a consequence of the librdkafka communication scheme which holds a connection open for each broker in the cluster regardless if you are producing to that broker or not.
